### ISSUE TITLE
High Memory Usage and Slow Startup on Linux

### ISSUE THREAD

**Issue by ForLoveOfCats** [JimmyCushnie/TUNG#35]
**OS:** Manjaro GNU/Linux x64 w/ Plasma 5
**TUNG Version:** 0.2.4
**Mods or Addons:** None

TUNG uses 1.29 Gigabytes of RAM sitting on the main menu.
![image](https://user-images.githubusercontent.com/30666851/38437806-9e690a9c-39a6-11e8-8379-72b7b00e0a14.png)
This causes excessive swapping on system with smaller amounts of RAM. Perhaps the long game startup time on Linux is related to this?

Cheers!

**Comment by pipe01** [comment]
785 MB on main menu for me on Ubuntu 16.04, compared to about 200 MB on Windows.

**Comment by ForLoveOfCats** [comment]
0.2.5 now uses between 2.7 to 3 gigs of ram on a relatively simple world. In addition to this it fires up over 50 threads (Is this normal?). World download below.

[Data Storage.zip](https://github.com/Iamsodarncool/TUNG/files/1887962/Data.Storage.zip)

**Comment by JimmyCushnie** [comment]
> In addition to this it fires up over 50 threads (Is this normal?)

No. Crap. I screwed up, 0.2.5's new system for colored component pieces is sucking up threads waaaaay more than it needs to. Looks like I'm doing a 0.2.6.

**Comment by JimmyCushnie** [comment]
This is absolutely baffling. I too am getting memory usage that's too high, but when I profile the game it tells me that <200MB are being used, so there's no way to tell where all that extra memory is going.
![image](https://user-images.githubusercontent.com/32693298/38510748-f3773ac8-3bf3-11e8-8698-e8cfbaa734b6.png)

**Comment by ForLoveOfCats** [comment]
### This comment is intended to correct information stated earlier as well as add information already discussed via other communication methods.

* It has been determined that the performance drop in 0.2.5 was not related to high memory usage. Instead it was a cross-platform issue regarding GPU utilization.
* It has been determined that the high thread count (50+) appears to be normal and having no ill effects to overall performance.
* 0.2.6 has lowered RAM usage on Linux slightly but is still rather high.

#### End of this comment

**Comment by ForLoveOfCats** [comment]
I have determined that the high memory usage is not caused by a memory leak as the absurdly high memory usage is allocated on game boot up and all actions taking place after that point do not raise memory usage significantly.
